src/realm/table.hpp

```diff
@@ -1641,6 +1643,19 @@ inline bool Table::has_shared_type() const noexcept
     return !m_top.is_attached();
 }
 
+inline void Table::verify_column(size_t col_ndx, const ColumnBase* col) const
+{
+    if (col_ndx < m_cols.size()) {
```

Comment:
This test isn't right. If you create a query that uses the last column in a table, then remove the first column in the table, using the query will now throw `column_does_not_exist`.